### ISSUE TITLE
QueryBuilder improvement. Add possibility to add statement after SELECT and column list

### ISSUE THREAD

**Issue by Patricy** [yiisoft/yii2#12430]
### What is the problem?

There is **$selectOption** property (string) in yii\db\Query, which appears between SELECT statement and list of columns in result sql query.

There is possibility in  MSSQL and PostgreSQL to add keywords _after_ list of columns, before FROM keyword.

My personal case: 
we doing pretty complicated query to export data, but related project needs to simply use result dataset in own processes. We running on MSSQL server, in secondary project I call model from primary and get Query object from there. I need to export all that data into temp db table.
Right now I need to make tricks (get raw SQL, hack it  to insert "_INTO consolidated_data_table_name_" statement between SELECT and FROM, execute it after)

Doc:
MSSDN: https://msdn.microsoft.com/en-us/library/ms189499.aspx
postgre: https://www.postgresql.org/docs/9.1/static/sql-selectinto.html

(I read discussion here about INSERT INTO SELECT feature, it little bit different,  it require result table to exists)
### What is the expected result?

I think - its good idea to have option to add anything after column list

Example:

```
$query->preSelectOption = "SQL_CALC_FOUND_ROWS";
$query->postSelectOption = "INTO consolidated_data_table_name";
```
### What do we have  instead?

`$query->selectOption = "SQL_CALC_FOUND_ROWS";`
only
### Additional info

Class
`yii\db\QueryBuilder`
Method
`public function buildSelect($columns, &$params, $distinct = false, $selectOption = null)`

I propose to
- rename parameter $selectOption  to $preSelectOption 
- add one more string parameter $postSelectOption = null

Method declaration change:
`public function buildSelect($columns, &$params, $distinct = false, $preSelectOption = null, $postSelectOption = null)`

Code change:

```
if ($selectOption !== null) {
     $select .= ' ' . $selectOption;
 }
```

to

```
if ($preSelectOption !== null) {
     $select .= ' ' . $preSelectOption;
 }
```

And at the end of method
`return $select . ' ' . implode(', ', $columns);`
change to

```
$select  .=  ' ' . implode(', ', $columns);
if ($postSelectOption !== null) {
    $select .= ' ' . $postSelectOption;
}
return $select;
```

By docs it related to MSSQL and PostgreSQL but I think in MySQL you can also get the case, when you do not want auto escaping columns, maybe some functions or etc.

Can I do pull request for this issue?

**Comment by CedricYii** [comment]
A generic way to handle this type of database specificities has been proposed in https://github.com/yiisoft/yii2/issues/10730

**Comment by samdark** [comment]
Won't be implemented in 2.0.
